appservice/src/utils/cpUtils.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { localize } from '../localize';
+
+export namespace cpUtils {
+    export async function executeCommand(command: string, commandOptions?: ICommandOptions, ...args: string[]): Promise<string> {
+        const outputChannel: vscode.OutputChannel | undefined = commandOptions && commandOptions.outputChannel ? commandOptions.outputChannel : undefined;
+        const result: ICommandResult = await tryExecuteCommand(command, commandOptions, ...args);
+        // command is only being used to output at this point so we can alter it
+        command = obfuscateValue(command, commandOptions);
+        if (result.code !== 0) {
+            // We want to make sure the full error message is displayed to the user, not just the error code.
+            // If outputChannel is defined, then we simply call 'outputChannel.show()' and throw a generic error telling the user to check the output window
+            // If outputChannel is _not_ defined, then we include the command's output in the error itself and rely on AzureActionHandler to display it properly
+            if (outputChannel) {
+                outputChannel.show();
+                throw new Error(localize('commandErrorWithOutput', 'Failed to run "{0}" command. Check output window for more details.', command));
```

Comment:
This error illustrates at least one reason why it's better to pass in "git" as the command and the args separately. The message would be 

> Failed to run "git" command

vs.

> Failed to run "git push -f ...... HEAD:master" command

...which probably has too much text for the notification